src/main/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -48,65 +58,104 @@
-
 	private final CloudFoundry2630AndLaterTaskLauncher taskLauncher;
-
-	protected static final Log logger = LogFactory.getLog(CloudFoundryAppScheduler.class);
+	private final CloudFoundrySchedulerProperties schedulerProperties;
 
 	public CloudFoundryAppScheduler(SchedulerClient client, CloudFoundryOperations operations,
-			CloudFoundryConnectionProperties properties, CloudFoundry2630AndLaterTaskLauncher taskLauncher) {
+			CloudFoundryConnectionProperties properties, CloudFoundry2630AndLaterTaskLauncher taskLauncher,
+			CloudFoundrySchedulerProperties schedulerProperties) {
 		Assert.notNull(client, "client must not be null");
 		Assert.notNull(operations, "operations must not be null");
 		Assert.notNull(properties, "properties must not be null");
 		Assert.notNull(taskLauncher, "taskLauncher must not be null");
+		Assert.notNull(schedulerProperties, "schedulerProperties must not be null");
 
 		this.client = client;
 		this.operations = operations;
 		this.properties = properties;
 		this.taskLauncher = taskLauncher;
+		this.schedulerProperties = schedulerProperties;
 	}
 
 	@Override
 	public void schedule(ScheduleRequest scheduleRequest) {
 		String appName = scheduleRequest.getDefinition().getName();
 		String jobName = scheduleRequest.getScheduleName();
+		logger.debug(String.format("Scheduling: ", jobName));
+
 		String command = stageTask(scheduleRequest);
 
 		String cronExpression = scheduleRequest.getSchedulerProperties().get(SchedulerPropertyKeys.CRON_EXPRESSION);
 		Assert.hasText(cronExpression, String.format(
 				"request's scheduleProperties must have a %s that is not null nor empty",
 				SchedulerPropertyKeys.CRON_EXPRESSION));
-		scheduleJob(appName, jobName, cronExpression, command);
+		int retryCount = 0;
+		boolean scheduleSuccessFlag = false;
+		while (!scheduleSuccessFlag &&
+				retryCount < this.schedulerProperties.getScheduleSSLRetryCount()) {
+			try {
+				retryCount++;
+				scheduleJob(appName, jobName, cronExpression, command);
+				scheduleSuccessFlag = true;
+			}
+			catch (CloudFoundryScheduleSSLException sle) {
+				logger.error(String.format(
+						"Failed to Schedule Task will retry. Attempt %d of %d",
+						retryCount,
+						this.schedulerProperties.getScheduleSSLRetryCount()));
+			}
+		}
 	}
 
 	@Override
 	public void unschedule(String scheduleName) {
-		throw new UnsupportedOperationException("Interface is not implemented for unschedule method.");
+		logger.debug(String.format("Unscheduling: ", scheduleName));
+		this.client.jobs().delete(DeleteJobRequest.builder()
+				.jobId(getJob(scheduleName))
+				.build())
+				.block();
 	}
 
 	@Override
 	public List<ScheduleInfo> list(String taskDefinitionName) {
-		throw new UnsupportedOperationException("Interface is not implemented for list method.");
+		return list().stream().filter(scheduleInfo ->
+				scheduleInfo.getTaskDefinitionName().equals(taskDefinitionName))
+				.collect(Collectors.toList());
```

Comment:
When performance is a concern, this approach (using streams for filtering) is not ideal.  I'm not saying to refactor this, but if performance becomes a concern (when that list is large) refactoring this to use a `for` loop may be a better idea.